win32ss/user/ntuser/window.h

```diff
@@ -124,4 +124,21 @@ BOOL FASTCALL IntBroadcastImeShowStatusChange(PWND pImeWnd, BOOL bShow);
 VOID FASTCALL IntNotifyImeShowStatus(PWND pImeWnd);
 VOID FASTCALL IntCheckImeShowStatusInThread(PWND pImeWnd);
 
+static inline
+VOID
+WndSetOwner(PWND pwnd, PWND pwndOwner)
+{
+    if (pwnd->spwndOwner != NULL)
+    {
+        UserDereferenceObject(pwnd->spwndOwner);
+    }
+
+    if (pwndOwner != NULL)
+    {
+        UserReferenceObject(pwndOwner);
```

Comment:
I May be paranoid here but please reference before dereferencing. In case that's the same window, this might lead it to be freed.